proxy/interceptor.go

```diff
@@ -98,6 +99,28 @@ func NewPayloadVisitorInterceptor(options PayloadVisitorInterceptorOptions) (grp
 
 		err := invoker(ctx, method, req, response, cc, opts...)
 		if err != nil {
+			if options.Inbound != nil {
+				stat, ok := status.FromError(err)
+				if ok {
+					// user provided payloads can sometimes end up in the status details of
+					// gRPC errors, make sure to visit those as well
+					newStatus := status.New(stat.Code(), stat.Message())
+					for _, detail := range stat.Details() {
+						detailAny, ok := detail.(*anypb.Any)
+						if ok && detailAny.MessageName() == "temporal.api.errordetails.v1.QueryFailedFailure" || detailAny.MessageName() == "temporal.api.errordetails.v1.MultiOperationExecutionFailure" {
```

Comment:
```suggestion
						if ok && (detailAny.MessageName() == "temporal.api.errordetails.v1.QueryFailedFailure" || detailAny.MessageName() == "temporal.api.errordetails.v1.MultiOperationExecutionFailure") {
```

Can we add parens when changing the comparison operator here?